presto-main/src/main/java/io/prestosql/operator/scalar/timestamp/HumanReadableSeconds.java

```diff
@@ -0,0 +1,136 @@
+    private static final String SECOND = "second";
+
+    private HumanReadableSeconds() {}
+
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice humanReadable(@SqlType(StandardTypes.DOUBLE) double inputSeconds)
+    {
+        long seconds = Double.valueOf(inputSeconds).longValue();
+
+        long weeks = processWeeks(seconds);
+        if (weeks > 0) {
+            seconds -= TimeUnit.DAYS.toSeconds(weeks * DAYS_IN_WEEK);
+        }
+
+        long days = processDays(seconds);
+        seconds -= TimeUnit.DAYS.toSeconds(days);
+
+        long hours = processHours(seconds);
+        seconds -= TimeUnit.HOURS.toSeconds(hours);
+
+        long minutes = processMinutes(seconds);
+        seconds -= TimeUnit.MINUTES.toSeconds(minutes);
+
+        long lseconds = processSeconds(seconds);
+
+        return getTimePeriods(weeks, days, hours, minutes, lseconds);
+    }
+
+    private static long processWeeks(@SqlType(StandardTypes.BIGINT) long seconds)
+    {
+        long weeks = 0;
+        if (seconds > 0) {
+            weeks = TimeUnit.SECONDS.toDays(seconds) / DAYS_IN_WEEK;
+        }
+        return weeks;
+    }
+
+    private static long processDays(@SqlType(StandardTypes.BIGINT) long seconds)
+    {
+        long days = 0;
+        if (seconds > 0) {
+            days = TimeUnit.SECONDS.toDays(seconds);
+        }
+        return days;
+    }
+
+    private static long processHours(@SqlType(StandardTypes.BIGINT) long remainingSeconds)
+    {
+        long hours = 0;
+        if (remainingSeconds > 0) {
+            hours = TimeUnit.SECONDS.toHours(remainingSeconds);
+        }
+        return hours;
+    }
+
+    private static long processMinutes(@SqlType(StandardTypes.BIGINT) long remainingSeconds)
+    {
+        long minutes = 0;
+        if (remainingSeconds > 0) {
+            minutes = TimeUnit.SECONDS.toMinutes(remainingSeconds);
+        }
+        return minutes;
+    }
+
+    private static long processSeconds(@SqlType(StandardTypes.BIGINT) long remainingSeconds)
+    {
+        long inputSeconds = 0;
+        if (remainingSeconds > 0) {
+            inputSeconds = remainingSeconds;
+        }
+        return inputSeconds;
+    }
+
+    private static Slice getTimePeriods(long weeks, long days, long hours, long minutes, long seconds)
+    {
+        StringJoiner stringJoiner = new StringJoiner(", ");
+        if (weeks > 0) {
+            stringJoiner.add(getPeriodType(weeks, WEEK));
+        }
+        if (days > 0) {
+            stringJoiner.add(getPeriodType(days, DAY));
+        }
+        if (hours > 0) {
+            stringJoiner.add(getPeriodType(hours, HOUR));
+        }
+        if (minutes > 0) {
+            stringJoiner.add(getPeriodType(minutes, MINUTE));
+        }
+        if (seconds > 0) {
+            stringJoiner.add(getPeriodType(seconds, SECOND));
+        }
+        return utf8Slice(stringJoiner.toString());
+    }
+
+    private static String getPeriodType(long val, String type)
```

Comment:
The name of this method doesn't describe what it does very accurately. I'd rename it to "render" or something similar